### PR TITLE
[v6r14] Better report from timeThis decorator

### PR DESCRIPTION
**Pull request by fstagni**
Also I took the opportunity to put some order in the unit tests of the Utilities

### CODE REVIEW

File: Core/Utilities/Time.py [hunk 1]
```diff
@@ -50,10 +50,17 @@ def timed( *args, **kw ):
     result = method( *args, **kw )
     te = nativetime.time()
 
+    pre = dt.utcnow().strftime( "%Y-%m-%d %H:%M:%S " ) + 'UTC '
```
Comment (phicharp) [inline Core/Utilities/Time.py]: you can also put the UTC in the format ;-)

File: Core/Utilities/Time.py [hunk 2]
```diff
@@ -50,10 +50,17 @@ def timed( *args, **kw ):
...
+      pre += args[0].log._systemName + '/' + args[0].log._subName + '   TIME: ' + args[0].transString
+    except AttributeError:
+      try:
+        pre += args[0].log._systemName + '    TIME: ' + args[0].transString
+      except AttributeError:
+        pre += args[0].log._systemName + '/' + args[0].log._subName + '   TIME: '
+    except IndexError:
+      pre += '  TIME : '
 
     print( "%s Exec time ===> function %r arguments len: %d -> %2.2f sec" % ( pre, method.__name__, len( kw ), te - ts ) )
```
Comment (phicharp) [inline Core/Utilities/Time.py]: 'arguments len' is not really explanatory... what about 
'(nb of named args: %d)' 
or 
'(%s named args)' % len(kw) if kw else 'no'
named to also be keyword if you prefer ;-)
What would be more interesting is the number of items the method is dealing with but we have no convention that the subject of the method is for example the first argument... Could we have a way to tell timeThis() how many "items" (i.e. files in most cases) the method is working on?

File: Core/Utilities/test/Test_Pfn.py [hunk 3]
```diff
@@ -18,17 +16,14 @@
...
-# @brief Definition of PfnTestCase class.
 
 ## imports 
 import unittest
-from DIRAC import S_OK, S_ERROR
-from DIRAC.Core.Utilities.Pfn import pfnparse, pfnunparse
-import random, string, timeit
 
+# sut
+from DIRAC.Core.Utilities.Pfn import pfnparse, pfnunparse
```
Comment (phicharp) [inline Core/Utilities/test/Test_Pfn.py]: these methods are deprecated if ever used

**Comment by phicharp** [comment]
See few remarks above, otherwise review OK!

File: Core/Utilities/test/Test_Pfn.py (hunk 3 above)
Comment (fstagni) [inline Core/Utilities/test/Test_Pfn.py]: They are actually used in several places, mostly in Resources/Storage

**Comment by fstagni** [comment]
Remarks taken into account. timeThis will now report on the length of the FIRST argument, if given (this is so-so... )

**Comment by phicharp** [comment]
Review OK

**Pull request merged**